RecoVertex/PrimaryVertexProducer/interface/DAClusterizerInZT_vect.h

```diff
@@ -0,0 +1,245 @@
+#ifndef DAClusterizerInZT_vect_h
+#define DAClusterizerInZT_vect_h
+
+/**\class DAClusterizerInZT_vect
+
+ Description: separates event tracks into clusters along the beam line
+
+	Version which auto-vectorizes with gcc 4.6 or newer
+
+ */
+
+#include "RecoVertex/PrimaryVertexProducer/interface/TrackClusterizerInZ.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrack.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include <vector>
+#include "DataFormats/Math/interface/Error.h"
+#include "RecoVertex/VertexTools/interface/VertexDistanceXY.h"
+#include "RecoVertex/VertexPrimitives/interface/TransientVertex.h"
+
+#include <memory>
+
+class DAClusterizerInZT_vect  final : public TrackClusterizerInZ {
+
+public:
+  
+  // Internal data structure to 
+  struct track_t {
+    
+    void AddItem( double new_z, double new_t, double new_dz2, double new_dt2, const reco::TransientTrack* new_tt, double new_pi   )
+    {
+      z.push_back( new_z );
+      t.push_back( new_t );
+      dz2.push_back( new_dz2 );
+      dt2.push_back( new_dt2 );
+      errsum.push_back( 1./(1./new_dz2 + 1./new_dt2) );
+      tt.push_back( new_tt );
+      
+      pi.push_back( new_pi ); // track weight
+      Z_sum.push_back( 1.0 ); // Z[i]   for DA clustering, initial value as done in ::fill
+    }
+
+    
+    
+    unsigned int GetSize() const
+    {
+      return z.size();
+    }
+    
+    
+    // has to be called everytime the items are modified
+    void ExtractRaw()
+    {
+      _z = &z.front();
+      _t = &t.front();
+      _dz2 = &dz2.front();
+      _dt2 = &dt2.front();
+      _errsum = &errsum.front();
+      _Z_sum = &Z_sum.front();
+      _pi = &pi.front();
+    }
+    
+    double * __restrict__ _z; // z-coordinate at point of closest approach to the beamline
```

Comment:
@Dr15Jones : is  ```__restrict__``` supported with all our compilers?

@lgray did you see that it really helps?

.. leading underscore in the names should not be used, by style restrictions

Reply:
I suppose to many of the style comments the answer should be: this is a copy-paste from DAClusterizerInZ_vect and there is a plan to merge the two in the nearest future.

Reply:
It is hard to tell, but I think Intel's icc compiler only allows `restrict` and then only if one uses the `--restrict` compiler option.
We HIGHLY discourage the use compiler specific directives. In this case, I'd suggest creating a macro RESTRICT_PTR which is turned on only for gcc and clang.

Reply:
there is no need of restrict (that, btw, do not necessarily work)
use 
```
#pragma GCC ivdep
```
or
OMP SIMD directives